bot/cogs/free.py

```
@@ -0,0 +1,123 @@
+
+        embed = Embed()
+        embed.colour = Colour.blurple()
+        embed.title = "**Looking for a free help channel?**"
+
+        if user is not None:
+            embed.description = f"**Hey {user.mention}!**\n\n"
+        else:
+            embed.description = ""
+
+        # Display all potentially inactive channels
+        # in descending order of inactivity
+        if free_channels:
+            embed.description += "**The following channel{0} look{1} free:**\n\n**".format(
+                's' if len(free_channels) > 1 else '',
+                '' if len(free_channels) > 1 else 's'
+            )
+
+            # Sort channels in descending order by seconds
+            # Get position in list, inactivity, and channel object
+            # For each channel, add to embed.description
+            for i, (inactive, channel) in enumerate(sorted(free_channels, reverse=True), 1):
+                minutes, seconds = divmod(inactive, 60)
+                if minutes > 59:
+                    hours, minutes = divmod(minutes, 60)
+                    embed.description += f"{i}. {channel.mention} inactive for {hours}h{minutes}m{seconds}s\n\n"
+                else:
+                    embed.description += f"{i}. {channel.mention} inactive for {minutes}m{seconds}s\n\n"
+
+            embed.description += ("**\nThese channels aren't guaranteed to be free, "
+                                  "so use your best judgement and check for yourself.")
+        else:
+            embed.description = ("**Doesn't look like any channels are available right now. "
+                                 "You're welcome to check for yourself to be sure. "
+                                 "If all channels are truly busy, please be patient "
+                                 "as one will likely be available soon.**")
+
+        await ctx.send(embed=embed)
+
+    @free.error
+    async def free_error(self, ctx: Context, error):
```

Because there's a local error handler, our global handler ignores any errors coming out of this cog:

```
Jan 09 18:53:48 pd.beardfist.com Bot: |                bot.cogs.events |    DEBUG | Command free has a local error handler, ignoring.
```

I'm not sure if reraising uncaught errors will help here, but they at least need to be logged.

---

This was my initial thought as well, however scragly told me to ignore it and not bother logging the error. Perhaps we still need to discuss this further.

---

It needs to at least be debug logged, otherwise we can't debug the cog without suppressing  the error handler.